x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/local/LocalSupplier.java

```diff
@@ -38,6 +48,23 @@ public Block[] get() {
             public String toString() {
                 return Arrays.toString(blocks);
             }
+
+            @Override
+            public void writeTo(StreamOutput out) throws IOException {
+                out.writeArray((o, v) -> ((PlanStreamOutput) o).writeBlock(v), blocks);
+            }
         };
     }
+
+    static LocalSupplier readFrom(PlanStreamInput in) throws IOException {
+        int count = in.readArraySize();
+        if (count == 0) {
+            return EMPTY;
+        }
+        Block[] blocks = new Block[count];
+        for (int i = 0; i < blocks.length; i++) {
+            blocks[i] = in.readBlock();
```

Comment:
I know we don't track memory here, but releasing blocks in case of exceptions would make our life easier in the future.